pandas/_testing/asserters.py

```diff
@@ -739,6 +745,23 @@ def assert_extension_array_equal(
     >>> b, c = a.array, a.array
     >>> tm.assert_extension_array_equal(b, c)
     """
+    if (
+        check_exact is lib.no_default
+        and rtol is lib.no_default
+        and atol is lib.no_default
+    ):
+        check_exact = (
+            is_numeric_dtype(left.dtype)
+            and not is_float_dtype(left.dtype)
+            or is_numeric_dtype(right.dtype)
+            and not is_float_dtype(right.dtype)
+        )
```

Comment:
nitpick - I can never remember how a logic statement like this will be parsed, fancy making it move explicit with
```python
        check_exact = (
            (is_numeric_dtype(left.dtype)
            and not is_float_dtype(left.dtype))
            or (is_numeric_dtype(right.dtype)
            and not is_float_dtype(right.dtype))
        )
```
?

Reply:
This would make the intendation worse? Not sure how black treats those

Reply:
up to you, but it's very easy to get these wrong, e.g. https://github.com/pandas-dev/pandas/pull/34334/files

Reply:
The intents look really off with parenthesis, will merge for now